src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -26,6 +26,15 @@ namespace ErrorCodes
     extern const int UNKNOWN_FORMAT_VERSION;
 };
 
+const CompressionCodecPtr & GinIndexCompression::zstdCodec()
+{
+    static constexpr auto GIN_COMPRESSION_CODEC = "ZSTD";
+    static constexpr auto GIN_COMPRESSION_LEVEL = 1;
+
+    static const auto & codec = CompressionCodecFactory::instance().get(GIN_COMPRESSION_CODEC, GIN_COMPRESSION_LEVEL);
```

Comment:
`CompressionCodecFactory::instance().get()` returns a `shared_ptr`. I'm not too sure about its lifetime if we capture it by reference. In case of doubt, capture it by value (i.e. increment the internal refcount).

Reply:
Interesting. Okay then capture by value seems better :)